spec/classes/foreman_proxy__plugin__container_gateway_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+
+describe 'foreman_proxy::plugin::container_gateway' do
+  on_plugin_os.each do |os, os_facts|
+    context "on #{os}" do
+      let(:facts) { os_facts }
+      let(:pre_condition) { 'include foreman_proxy' }
+
+      describe 'with default settings' do
+        it { should contain_foreman_proxy__plugin__module('container_gateway') }
+        it 'container_gateway.yml should contain the correct configuration' do
+          verify_exact_contents(catalogue, '/etc/foreman-proxy/settings.d/container_gateway.yml', [
+            '---',
+            ':enabled: https',
+            ":pulp_endpoint: https://#{facts[:fqdn]}",
+            ':pulp_client_ssl_cert: /etc/foreman-proxy/foreman_ssl_cert.pem',
```

Comment:
@ekohl is there a way to "mock" a variable from another class?  It seems like `$foreman_proxy::foreman_ssl_cert` and key is `undef` for the spec tests.

I'm hitting:
```
     Puppet::PreformattedError:
       Evaluation Error: Error while evaluating a Function Call, Class[Foreman_proxy::Plugin::Container_gateway]:
         parameter 'pulp_client_ssl_cert' expects a Stdlib::Absolutepath = Variant[Stdlib::Windowspath = Pattern[/\A(([a-zA-Z]:[\\\/])|([\\\/][\\\/][^\\\/]+[\\\/][^\\\/]+)|([\\\/][\\\/]\?[\\\/][^\\\/]+)).*\z/], Stdlib::Unixpath = Pattern[/\A\/([^\n\/\0]+\/*)*\z/]] value, got Undef
         parameter 'pulp_client_ssl_key' expects a Stdlib::Absolutepath = Variant[Stdlib::Windowspath = Pattern[/\A(([a-zA-Z]:[\\\/])|([\\\/][\\\/][^\\\/]+[\\\/][^\\\/]+)|([\\\/][\\\/]\?[\\\/][^\\\/]+)).*\z/], Stdlib::Unixpath = Pattern[/\A\/([^\n\/\0]+\/*)*\z/]] value, got Undef (line: 3, column: 1) on node cannolo.usersys.redhat.com
```

Reply:
There is
```puppet
let(:pre_condition) { 'include foreman_proxy' }
```

Reply:
I do have that.  I think it's reporting `undef` because of this: https://github.com/theforeman/puppet-foreman_proxy/blob/e4b7763e45fecc5fbaa5d88cc066906d6caf7bf7/manifests/params.pp#L172

Perhaps I can override the parameter itself?

Reply:
Okay, looks like I figured it out. I updated the test in my latest push.